src/meta/src/hummock/manager/tests.rs

```diff
@@ -2139,55 +2292,43 @@ async fn test_partition_level() {
     let (env, hummock_manager, _, worker_node) =
         setup_compute_env_with_metric(80, config.clone(), Some(MetaMetrics::for_test(&registry)))
             .await;
-    let context_id = worker_node.id;
-
-    hummock_manager
-        .register_table_ids_for_test(&[(100, 2)])
-        .await
-        .unwrap();
+    let hummock_meta_client: Arc<dyn HummockMetaClient> = Arc::new(MockHummockMetaClient::new(
+        hummock_manager.clone(),
+        worker_node.id,
+    ));
     hummock_manager
-        .register_table_ids_for_test(&[(101, 2)])
+        .register_table_ids_for_test(&[(100, 2), (101, 2)])
         .await
         .unwrap();
-    let sst_1 = gen_local_sstable_info(10, 1, vec![100, 101]);
-    hummock_manager
-        .commit_epoch_for_test(30, vec![sst_1.clone()], HashMap::from([(10, context_id)]))
-        .await
-        .unwrap();
-    // Construct data via manual compaction
-    let compaction_task = get_manual_compact_task(&hummock_manager, context_id).await;
-    let base_level: usize = 6;
-    assert_eq!(compaction_task.input_ssts[0].table_infos.len(), 1);
-    assert_eq!(compaction_task.target_level, base_level as u32);
-    assert!(hummock_manager
-        .report_compact_task(
-            compaction_task.task_id,
-            TaskStatus::Success,
-            vec![
-                gen_sstable_info(11, 1, vec![100]),
-                gen_sstable_info(12, 2, vec![101]),
-            ],
-            None,
+    let mut sst_1 = gen_local_sstable_info(10, 1, vec![100, 101]);
+    {
+        let epoch = test_epoch(1);
+        let table_key_l = gen_key_from_str(VirtualNode::from_index(1), "1");
+        let table_key_r = gen_key_from_str(VirtualNode::from_index(128), "1");
+        let full_key_l = FullKey::for_test(TableId::new(100), table_key_l, epoch).encode();
+        let full_key_r = FullKey::for_test(TableId::new(101), table_key_r, epoch).encode();
+        sst_1.sst_info.key_range.left = full_key_l.into();
+        sst_1.sst_info.key_range.right = full_key_r.into();
```

Comment:
I think the reason why we modify the key range here is because  `gen_local_sstable_info` (more accurately [gen_sstable_info](https://github.com/risingwavelabs/risingwave/blob/d62753ae1bb508ee05aeffb8f840bc69fb619bf7/src/meta/src/hummock/manager/tests.rs#L65)) is filling the key range with `iterator_test_key_of_epoch(1, idx, 1)`, which doesn't aligh with the provided `table_ids`. Is it possible to change the implementation of `gen_sstable_info` to fill in the correct key range based on `table_ids`? We can have a util function similar to the logic here to generate valid full key range based on provided `table_ids`.

Reply:
Thanks, I've fixed this logic in a new commit, so let's change it step by step. I even found some uts that relied on this wrong `key_range` to pass check, so I'm determined to change it .....